src/CausationMetadataEnricher.php

```diff
@@ -57,7 +62,7 @@ function (ActionEvent $event): void {
             1000
         );
 
-        $eventEmitter->attachListener(
+        $this->eventStoreListeners[] = $eventStore->attach(
             ActionEventEmitterEventStore::EVENT_CREATE,
             function (ActionEvent $event): void {
                 if (null === $this->currentCommand || ! $this->currentCommand instanceof Message) {
```

Comment:
`null` is also NOT an instance of `Message`, so this is a doubled expression I guess. Not sure about performance here.